src/MultiIndices/FixedMultiIndexSet.cpp

```diff
@@ -299,6 +319,27 @@ int FixedMultiIndexSet<MemorySpace>::MultiToIndex(std::vector<unsigned int> cons
     }
 }
 
+template<typename MemorySpace>
+MultiIndexSet FixedMultiIndexSet<MemorySpace>::Unfix() const
+{
+    if(!isCompressed)
+        throw std::runtime_error("MultiIndexSet must be compressed to unfix");
+    Kokkos::View<unsigned int*, Kokkos::HostSpace> h_nzStarts = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), nzStarts);
+    Kokkos::View<unsigned int*, Kokkos::HostSpace> h_nzDims = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), nzDims);
+    Kokkos::View<unsigned int*, Kokkos::HostSpace> h_nzOrders = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), nzOrders);
+    MultiIndexSet output (this->dim);
+    for(int term = 0; term < h_nzStarts.extent(0)-1; term++){
+        unsigned int start = h_nzStarts(term);
+        unsigned int end = h_nzStarts(term+1);
+        unsigned int numNz = end - start;
+        unsigned int* nzIndTerm = h_nzDims.data() + start;
+        unsigned int* nzValTerm = h_nzOrders.data() + start;
+        MultiIndex midx_term {nzIndTerm, nzValTerm, numNz, this->dim};
```

Comment:
See comment above about using subviews instead of pointers, which I think would be safer without sacrificing much speed.